app/Http/Middleware/RegistrationValidation.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+
+class RegistrationValidation
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response|\Illuminate\Http\RedirectResponse)  $next
+     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        $request->validate([
+            'name' => 'required|max:20|regex:/^[a-zA-ZÑñ\s]+$/',
+            'email' => 'required|email|unique:users,email',
+            'password' => 'required|min:8',
+          ]);
```

Comment:
these things can be done in controller itself. We dont need middleware for validation